src/index.coffee

```diff
@@ -67,23 +67,44 @@ reinstall = (options = {}, pkg) ->
 
   return if pkg then curried pkg else curried
 
-discover = (path = '../package.json') ->
-  path = resolve path
-  delete require.cache[path]
-  { gitDependencies } = require path
+discover = (package_json = '../package.json') ->
+  package_json = resolve package_json
+  delete require.cache[package_json]
+  { gitDependencies } = require package_json
   ( url for name, url of gitDependencies)
 
+sha_for = (name_to_find, shrinkwrap_file = '../git-shrinkwrap.json') ->
+  shrinkwrap_file = resolve shrinkwrap_file
+  delete require.cache[shrinkwrap_file]
+  { dependencies } = require shrinkwrap_file
+  for name, url of dependencies
+    return url if name = name_to_find
+
 ###
 
 As seen on http://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html
 
 ###
 
-reinstall_all = (options = {}, packages) ->
+reinstall_all = (packages, options = {}) ->
+
   curried = (packages) ->
     factories = packages.map (url) ->
       [ whole, url, revision] = url.match /^(.+?)(?:#(.+))?$/
       revision ?= 'master'
+      name = url.split(':').pop().replace(/\.git$/, "")
+
+      try
```

Comment:
After some consideration I think that reading and writing configuration files ({package,git-shrinkwrap}.json) should be a concern of CLI or user's build system. I'll move it there, ok?